### DISCUSSION TITLE
Speedtest-tracker tests measure 100-150mbps slower than speedtest.net in browser

### DISCUSSION THREAD

**Discussion by oonqt** [alexjustesen/speedtest-tracker#1338] [Q&A]
I'm running in Docker on Windows. I noticed I haven't got a single speed test over 370Mbps with speedtest-tracker.  It doesn't matter if it's a scheduled test, or one that I manually started, or if I manually run a test via console in the container with `speedtest --accept-license --accept-gdpr`. In the browser, I get consistently between 400-500 (these are the speeds I pay for). I get these same results whether I run speedtest.net test on a separate computer, or on my Windows server running Docker. Could this have something to do with the overhead when using Docker networking? That's the only thing that I can think of. Any help is appreciated.

**Comment by oonqt** [comment]
After further investigation, I can see my CPU utilization is 100%. This looks like a bottleneck, and I'm almost certain it's because of Docker for Windows. Networking isn't nearly as good as it is on Linux. I'm not really sure what to do now as there doesn't appear to be a native windows version.

**Reply by alexjustesen** [reply]
This is correct, Docker for Windows does have network bottlenecks which can cause those speeds. I see them as well locally when doing development even in WSL.

You can always get a cheap Raspberry Pi 4B which can saturate a 1Gbps network.

**Comment by sobeservices** [comment]
I am receiving near the same type of issue. I run Speedtracker from my raspberry pi on a Ubuntu 22 with a direct wired connection into the back of my home gateway. No mater if I am running the test manually from Speed tracker or have the automated cron run it either every 1hr or every 30 mins my speed test come out way lower using Speed Tracker then when I go from a typical mac computer or even running it off my AppleTV. The test are about 200 to 400 or more mbps lower. My average speed test running outside of Speedtracker is 800 to 900 and I have 1 gig service. So why is there so much discrepancy. I use AT&T servers for running the test and just simple Speed Tracker is just showing slower results which I don't understand.

Could someone help and explain. What am I doing wrong? or is this normal? The only thing that runs off my mini Ubuntu server is this program .

**Reply by alexjustesen** [reply]
I'd check this issue out and see if `eee` might be the cause of your problem. If you search though the issues you'll see various resolution for low speeds.

https://github.com/alexjustesen/speedtest-tracker/issues/1263#issuecomment-1971482601

**Reply by sobeservices** [reply]
I have tried all of these and still have the same issue. I tried to see if energy efficiency was the problem. My device wasn't even running it. Tried the offset. I tried everything listed here. Do you have any other suggestions?

Just noticed something I ran speedtest-cli and got the following.

Retrieving speedtest.net server list...
Selecting best server based on ping...
Hosted by Summit Broadband (XXXXXX) [162.56 km]: 15.271 ms
Testing download speed................................................................................
Download: 501.08 Mbit/s
Testing upload speed......................................................................................................
Upload: 93.87 Mbit/s
root@raspberrypi4-ubuntu:/home/administrator

Not good but still confused why this is giving me such slow results and anything else on my network is way higher in the 800 700's range wired with direct gateway connections no router switches or anything in the middle that would interfere. I even tried replacing the ethernet cable.

humm still puzzled
